### PR TITLE
Fix ComboBox onBlur when relatedTarget is an ancestor to menu

### LINKED ISSUE #16077: ComboBox - allowFreeform not triggering onChange when clicking outside (onblur event)
### Environment Information

- **Package version(s)**: office-ui-fabric-react@7.153.2

### Please provide a reproduction of the bug in a codepen:

My use case: https://codepen.io/marwankhalili/pen/rNMNWPw
Minimal test case: https://codepen.io/marwankhalili/pen/qBaBRgz

#### Actual behavior:

I type a value in the textfield and click outside on the grey background but `onChange` does not trigger. 

I re-enter the value and click outside on the white background, `onChange` triggers correctly.

#### Expected behavior:

`onChange` should trigger when clicking on the grey background (outside the ComboBox)

### Priorities and help requested:

Are you willing to submit a PR to fix? Yes

Requested priority: High

--------------------------------

This issue occurs when using `allowFreeform` and the `onblur` event triggers on an ancestor element to the ComboBox's list/dropdown menu. If `persistMenu` is disabled then the issue will only occur if the ComboBox menu is open when clicking outside. I enabled `persistMenu` in the CodePen examples for demonstration purposes since that causes the issue to occur even when the ComboBox menu is closed.

I detected this when I was working on a SharePoint WebPart. SharePoint wraps the entire WebPart in an element with a `tabindex` attribute which is caught by the `onblur` event. This means that `allowFreeform` does not to work reliably with mouse clicks in SharePoint WebPart solutions.

### PR DESCRIPTION
**Pull request by marwankhalili**
#### Pull request checklist

- [x] Addresses an existing issue: Fixes #16077
- [x] Include a change request file using `$ yarn change`

#### Description of changes

ComboBox onBlur function will now call `_submitPendingValue` if the event was triggered by an ancestor element to the ComboBox's dropdown menu.

Previously, the onBlur function would return without doing anything in this case. The intended behavior was to ignore onblur events triggered by ancestor elements that were located within the menu (e.g. scrollbar when the menu is too small), however it only checks that the `relatedTarget` element is an ancestor to the menu and not its location (see examples in #16077).

#### Focus areas to test

I've focused on visual regression tests for ComboBox with allowFreeform and triggering onBlur on related elements. Also tested IE11 compatibility.

### CODE REVIEW

**Review by Ridvanovskyy — APPROVED** [review]

**Comment by JustSlone** [comment]
@joschect we need code owner review on this PR, can you take a look soon?

**Review by joschect — APPROVED** [review]

**Comment by layershifter** [comment]
@JustSlone we have an approval from @joschect. Anything else before merging? 🐱

**Comment by JustSlone** [comment]
@layershifter nope, codeowner approval is enough

**Pull request merged**
